### DISCUSSION TITLE
How can i get input, output tensors with requires_grads set as True? And can i pre-specify bounding boxes?

### DISCUSSION THREAD

**Discussion by mshergad** [ultralytics/yolov5#11969] [Q&A]
Hello,
I am trying to run (explainability) XAI approaches on Yolov5 for which I need both the input and output tensor to store the gradient information. Is there an easy way for me to do so? The goal is to pick one specific detected object tensor for an input tensor with grad enabled, and then use the detected bounding boxes in the model's forward passes as prespecified input and get output tensors as I vary input tensors to do XAI analysis. 
Thanks!
